include/boost/url/pct_encoded_view.hpp

```diff
@@ -0,0 +1,696 @@
+//
+// Copyright (c) 2022 Alan Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_PCT_ENCODED_VIEW_HPP
+#define BOOST_URL_PCT_ENCODED_VIEW_HPP
+
+#include <boost/url/detail/config.hpp>
+#include <boost/url/string_view.hpp>
+#include <boost/url/pct_encoding.hpp>
+#include <type_traits>
+#include <iterator>
+#include <iosfwd>
+
+namespace boost {
+namespace urls {
+
+/** A view of the percent decoded characters in a string
+
+    This class is a lightweight wrapper that presents
+    a view of an underlying sequence of percent-encoded
+    characters mutating or copying it.
+
+    The reference objects do not own the underlying
+    character buffer, are inexpensive to create and
+    copy. This avoid any unnecessary memory allocations
+    when working with decoded strings, and transfers
+    this responsibility to the user.
```

Comment:
I think it would be better to just say that "Storage for the underlying character buffer is not transferred; the caller is responsible for ensuring that the lifetime of the buffer lasts until the view is destroyed."

In terms of complexity ("inexpensive") you should have "@par Complexity" clauses on the member functions and in the iterator javadoc